docs/contributing/backward-compatibility-promise.md

```diff
@@ -0,0 +1,144 @@
+Existing routes should not be modified, except for adding a new [optional placeholder](https://symfony.com/doc/3.4/routing/optional_placeholders.html).
+
+Route names may be changed only in a `MAJOR` release.
+
+### Docker Configuration and Orchestration Manifests
+Configuration of containers and orchestration is mostly in the `project-base` repository, which means that [it's excluded form the BC promise](#project-base-repository).
+
+The changes should always be described in [upgrade notes](/UPGRADE.md) (in the *Infrastructure* section).
+
+### Twig
+Twig blocks, functions, filters and the templates themselves can be removed or renamed only in a `MAJOR` release.
+
+Changes of Twig functions and filters in `MINOR` and `PATCH` releases must be backward-compatible.
+This means only a new optional argument or a support for new data type of existing argument may be added.
+
+### LESS / CSS
+Classes can be renamed only in a `MAJOR` release.
+
+Changes of the visual style of the administration are not considered a BC break and may be introduced in any version.
+
+The style of the front-end is in the `project-base` repository, which means that [it's excluded form the BC promise](#project-base-repository).
+
+### Javascript
+Javascript code should adhere to similar rules as the PHP code ([except in project-base repository](#project-base-repository)).
+
+Behavior of elements with classes prefixed with `.js-` should change only in a `MAJOR` release.
+New keys in a `options` map or a new expected data attribute may be added in a `MINOR` or `PATCH` release only if it has a default value.
+Adding a new optional argument to a method is considered a BC break (as it the method might be overridden by the user with the new argument already in use).
+
+## Summary
+
+### If You Are a User of Shopsys Framework...
+- we use [Semantic Versioning](http://semver.org/spec/v2.0.0.html) (`MAJOR.MINOR.PATCH`)
+- all higher `MINOR` and `PATCH` version should be compatible with your project
+- read the [section Using Symfony Code](https://symfony.com/doc/3.4/contributing/code/bc.html#using-symfony-code) in their BC Promise for a nice clarification about BC in PHP
+- use [caret version ranges](https://getcomposer.org/doc/articles/versions.md#caret-version-range-) in your `composer.json` (eg. `^7.0.0`)
+- watch for changes marked as `BC-BREAK` in the [change log](/CHANGELOG.md)
+- don't forget to execute new DB migrations via `./phing db-migrations` during upgrades
+- when upgrading to a new major release, read the [upgrade notes](/UPGRADE.md)
+- it's recommended to always upgrade to the highest minor version first, fix all deprecation notices, and then upgrade to the next major release
+- upgrade often, it will be easier
+
+### If You Are a Contributor to Shopsys Framework...
+- read about [Semantic Versioning](http://semver.org/spec/v2.0.0.html)
+- when making a change, always think about backward compatibility
+- instead of removing you can add a deprecation
```

Comment:
how? It would be useful to lead user to source "how to add deprecation"

Reply:
linked in 61a4a776a